scarlet/source.py

```diff
@@ -191,32 +198,12 @@ def _make_initial(self, center, img, shape, psf, config, tiny=1e-10):
         """
         B, Ny, Nx = img.shape
         _y, _x = center_int = np.round(center).astype('int')
-        if self.normalization != sc.Normalization.A:
-            sed = img[:, _y, _x]
-            if psf is not None:
-                # Increase the magnitude of the SED's to account
-                # for the normalized PSF's
-                py, px = (psf.shape[0] // 2, psf.shape[1] //2)
-                centers = psf[:, py, px]
-                centers = np.max(psf, axis=(1,2))
-                sed = sed/centers
-            sed = prox_sed_on(sed, 0)
-            morph = np.zeros(shape, dtype=img.dtype)
-            # Turn on a single pixel at the peak
-            cy, cx = (shape[0] // 2, shape[1] //2)
-            morph[cy, cx] = 1
-            return sed, morph
-        else:
-            # determine initial SED from peak position
-            try:
-                sed = get_pixel_sed(img, center_int)
-            except SourceInitError:
-                # flat weights as fall-back
-                sed = np.ones(B) / B
-            morph = np.zeros(shape)
-            # Turn on a single pixel at the peak
-            cy, cx = (shape[0] // 2, shape[1] //2)
-            morph[cy, cx] = max(img[:,_y,_x].sum(axis=0), tiny)
+        # determine initial SED from peak position: amplitude is in sed
+        sed = get_pixel_sed(img, center_int)
+        morph = np.zeros(shape)
```

Comment:
This should be `morph = np.zeros(shape, dtype=img.dtype)`. This is one part of the code that we need to clean up because the stack often uses 32 bit floats instead of doubles and at some point we'll need to go back through the code and find everywhere that we don't set the type.